docs/apworld dev faq.md

```diff
@@ -0,0 +1,26 @@
+# APWorld Dev FAQ
+
+This document is meant as a reference tool to show the best known solutions to common problems when developing an apworld
+
+## My game has a restrictive start that leads to fill errors
+
+
+Hint to the Generator that an item needs to be in sphere one with local_early_items
+```
+early_item_name = "Sword"
+self.multiworld.local_early_items[self.player][early_item_name] = 1
+```
+
+## I have multiple settings that change the item/location pool counts and need to balance them out
+
+
+Create extra filler based on the difference between your unfilled locations and your itempool by comparing [get_unfilled_locations](/BaseClasses.py) to your list of items to submit
```

Comment:
this link should link to the actual relevant part not the entire file. also due to the [sphinx docs migration](https://github.com/ArchipelagoMW/Archipelago/pull/2909) it should be a direct github link

Reply:
I didn't want to link to a line number (because they could change and it would be a horrid mess to maintain that) and couldn't figure out to link otherwise, do you have an example?

Reply:
found a filter that seems to work, but not always, and was noted as not supported by all browsers
I think I'll leave it with the filter because worst case it functions the same way as it did without the filter

Also made it a direct link, though once we get the sphinx docs I imagine we may want to link to those docs instead of the source code,